homeassistant/components/vicare/entity.py

```diff
@@ -29,13 +29,19 @@ def __init__(
         gateway_serial = device_config.getConfig().serial
         device_id = device_config.getId()
         model = device_config.getModel().replace("_", " ")
+        via_device_identifier: tuple[str, str]
 
         identifier = (
             f"{gateway_serial}_{device_serial.replace('-', '_')}"
             if device_serial is not None
             else f"{gateway_serial}_{device_id}"
         )
 
+        if device_serial is not None and device_serial.startswith("zigbee-"):
+            parts = device_serial.split("-")
+            if len(parts) == 3:  # expect format zigbee-<zigbee-ieee>-<channel-id>
```

Comment:
Should we log a warning if not 3?

Reply:
No, if not three, we have a normal zigbee device and proceed without a `via_device` link.